File: examples/matlab/producer_consumer/Consumer.m
function Consumer(mm_ip)

    global RTMA;

    RTMA_BaseDir = '../../..';
    addpath([RTMA_BaseDir '/lang/matlab']);

    MessageTypes =  { 'TEST_DATA' };

    ConnectArgs = {'CONSUMER', RTMA_BaseDir, ['./message_defs.mat']};
    if exist('mm_ip','var') && ~isempty(mm_ip)
        ConnectArgs{end+1} = ['-server_name ' mm_ip];
    end

    ConnectToMMM(ConnectArgs{:});
    Subscribe( MessageTypes{:})

    disp 'Consumer running..'

    while( 1)
        M = ReadMessage( 'blocking');
        switch(M.msg_type)
            case 'TEST_DATA'
                fprintf('Received message %s  Data = [a:%d  b:%d  x:%f]\n', M.msg_type, M.data.a, M.data.b, M.data.x);
        end
    end

    DisconnectFromMMM

File: examples/python/producer_consumer/Consumer.py
#!/usr/bin/python
from __future__ import print_function
import time
import PyRTMA
from PyRTMA import copy_from_msg
import message_defs as mdefs
import sys

MID_CONSUMER = 11

if __name__ == "__main__":
    mod = PyRTMA.RTMA_Module(MID_CONSUMER, 0)
    mod.ConnectToMMM("localhost:7111")
    mod.Subscribe(mdefs.MT_TEST_DATA)
    
    print("Consumer running...\n")
    
    while (1):
        msg = PyRTMA.CMessage()
        mod.ReadMessage(msg)    # blocking read
        print("Received message ", msg.GetHeader().msg_type)

        if msg.GetHeader().msg_type == mdefs.MT_TEST_DATA:
            msg_data = mdefs.MDF_TEST_DATA()
            copy_from_msg(msg_data, msg)
            print("  Data = [a: %d, b: %d, x: %f]" % (msg_data.a, msg_data.b, msg_data.x))
        
    mod.DisconnectFromMMM()
        
